importexport/musicxml/importmxmlpass2.cpp

```diff
@@ -473,13 +473,17 @@ static Instrument createInstrument(const MusicXMLInstrument& mxmlInstr, const In
 //   updatePartWithInstrument
 //---------------------------------------------------------
 
-static void updatePartWithInstrument(Part* const part, const MusicXMLInstrument& mxmlInstr, const Interval interval)
+static void updatePartWithInstrument(Part* const part, const MusicXMLInstrument& mxmlInstr, const Interval interval, const bool hasDrumset = false)
       {
-      const Instrument instr = createInstrument(mxmlInstr, interval);
+      Instrument instr = createInstrument(mxmlInstr, interval);
+      if (hasDrumset)
+            instr.channel(0)->setBank(128);
       part->setInstrument(instr);
-      if (mxmlInstr.midiChannel >= 0) part->setMidiChannel(mxmlInstr.midiChannel, mxmlInstr.midiPort);
+      if (mxmlInstr.midiChannel >= 0)
+            part->setMidiChannel(mxmlInstr.midiChannel, mxmlInstr.midiPort);
       // note: setMidiProgram() does more than simply setting the MIDI program
-      if (mxmlInstr.midiProgram >= 0) part->setMidiProgram(mxmlInstr.midiProgram);
+      if (mxmlInstr.midiProgram >= 0)
+            part->setMidiProgram(mxmlInstr.midiProgram);
```

Comment:
This doesn't seem to be an issue for this pull request, but `part->setMidiProgram()` also sets bank to a value which is 0 by default:
https://github.com/musescore/MuseScore/blob/4983887d71bbab2197e00717a2eec179a009d2c1/libmscore/part.h#L108

So with this PR `updatePartWithInstrument` effectively sets the channel's bank to 128, adds the channel to MIDI mapping (via `setMidiChannel()`) and then sets the channel's bank value back to 0. Here it doesn't seem to cause any issues (although it doesn't seem right anyway),  but in other places (for example, in [OVE import](https://github.com/musescore/MuseScore/blob/4983887d71bbab2197e00717a2eec179a009d2c1/importexport/ove/importove.cpp#L623-L624)) it might cause undesirable results. So maybe `setMidiProgram()` behavior should be somehow reconsidered, although this certainly doesn't belong to this PR.

Reply:
Thank you very much for your comment. 

So line `part->instrument()->channel(0)->setBank(128); ` is not actually redundant, in the end.